ecommerce/serializers.py

```diff
@@ -155,6 +159,49 @@ def get_coupons(self, instance):
             )
         ]
 
+    def get_data_consents(self, instance):
+        """ Get the DataConsentUser objects associated with the basket via coupon and product"""
+        data_consents = []
+        coupon_selections = models.CouponSelection.objects.filter(basket=instance)
+        if coupon_selections:
+            courses = [
+                course
+                for courselist in [
+                    get_product_courses(item.product)
+                    for item in instance.basketitems.all()
+                ]
+                for course in courselist
+            ]
+
+            for coupon_selection in coupon_selections:
+                company = latest_coupon_version(
+                    coupon_selection.coupon
+                ).payment_version.company
+                if company:
+                    agreements = (
+                        models.DataConsentAgreement.objects.select_related("company")
+                        .prefetch_related("courses")
```

Comment:
I'm not sure the `select_related` or `prefetch_related` are useful here since we don't have any code referencing foreign keys like `agreement.company` or `agreement.courses.all()` below.